pallets/connectors-gateway/routers/src/lib.rs

```diff
@@ -0,0 +1,352 @@
+		)
+		.map_err(|e| e.error)?;
+
+		Ok(())
+	}
+}
+
+/// The EVMDomain holds all relevant information for validating and executing
+/// the EVM call.
+#[derive(Debug, Encode, Decode, Clone, PartialEq, Eq, TypeInfo, MaxEncodedLen)]
+pub struct EVMDomain {
+	/// The address of the contract deployed in our EVM.
+	pub target_contract_address: H160,
+
+	/// The `BlakeTwo256` hash of the target contract code.
+	///
+	/// This is used during router initialization to ensure that the correct
+	/// contract code is used.
+	pub target_contract_hash: H256,
+
+	/// The values used when executing the EVM call.
+	pub fee_values: FeeValues,
+}
+
+/// The FeeValues holds all information related to the transaction costs.
+#[derive(Debug, Encode, Decode, Clone, PartialEq, Eq, TypeInfo, MaxEncodedLen)]
+pub struct FeeValues {
+	/// The value used when executing the EVM call.
+	pub value: U256,
+
+	/// The gas price used when executing the EVM call.
+	pub gas_price: U256,
+
+	/// The gas limit used when executing the EVM call.
+	pub gas_limit: U256,
+}
+
+/// A generic router used for executing XCM calls.
+#[derive(Debug, Encode, Decode, Clone, PartialEq, Eq, TypeInfo, MaxEncodedLen)]
+pub struct XCMRouter<T>
+where
+	T: frame_system::Config + pallet_xcm_transactor::Config,
+{
+	pub xcm_domain: XcmDomain<T::CurrencyId>,
+	pub _marker: PhantomData<T>,
+}
+
+impl<T> XCMRouter<T>
+where
+	T: frame_system::Config + pallet_xcm_transactor::Config,
+{
+	/// Sets the weight information for the provided XCM domain location, and
+	/// the fee per second for the provided fee asset location.
+	pub fn do_init(&self) -> DispatchResult {
+		pallet_xcm_transactor::Pallet::<T>::set_transact_info(
+			<T as frame_system::Config>::RuntimeOrigin::root(),
+			self.xcm_domain.location.clone(),
+			self.xcm_domain.transact_info.transact_extra_weight,
+			self.xcm_domain.transact_info.max_weight,
+			self.xcm_domain.transact_info.transact_extra_weight_signed,
+		)?;
+
+		pallet_xcm_transactor::Pallet::<T>::set_fee_per_second(
+			<T as frame_system::Config>::RuntimeOrigin::root(),
+			self.xcm_domain.fee_asset_location.clone(),
+			self.xcm_domain.fee_per_second,
+		)
+	}
+
+	/// Encodes the Connectors message to the required format and executes the
+	/// call via the XCM transactor pallet.
+	pub fn do_send(&self, sender: T::AccountId, msg: Vec<u8>) -> DispatchResult {
+		let ethereum_xcm_call = get_encoded_ethereum_xcm_call::<T>(self.xcm_domain.clone(), msg)
+			.map_err(|_| DispatchError::Other("encoded ethereum xcm call retrieval"))?;
+
+		pallet_xcm_transactor::Pallet::<T>::transact_through_sovereign(
+			<T as frame_system::Config>::RuntimeOrigin::root(),
+			// The destination to which the message should be sent.
+			self.xcm_domain.location.clone(),
+			// The sender will pay for this transaction.
+			sender,
+			// The currency in which we want to pay fees.
+			CurrencyPayment {
+				currency: Currency::AsCurrencyId(self.xcm_domain.fee_currency.clone()),
+				fee_amount: None,
```

Comment:
Why is this `None`?

Reply:
This was as per the original connectors implementation found here - https://github.com/centrifuge/centrifuge-chain/blob/ac7ee491cb805d99e97122a47d86c3951ce7b03b/pallets/connectors/src/lib.rs#L541

Maybe @NunoAlexandre can provide more specific info?